site/content/docs/authentication/handling-auth-yourself.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Handling Authentication Yourself (in Automation)"
+tags: 
+- authentication
+- guide
+type: page
+---
+
+If you can generate an authentication token (e.g. to use in a header or cookie) and you know that your app
+will not invalidate it while you are using ZAP then one option it to handle authentication yourself.
+
+In this case you take on the responsibility for handling the authentication and session handling.
+This means that you do not have to configure ZAP as much but it also means that ZAP will not be able to help you as much.
+
+### Authentication Env Vars
+
+If your app just needs one authentication token which you can generate outside of ZAP then you can use the 
+ZAP Authentication environmental variables.
+
+There are 3 of these env vars which are documented on the Desktop User Guide [Authentication](/docs/desktop/start/features/authentication/#envvars) page and reproduced here for completeness:
+
+* `ZAP_AUTH_HEADER_VALUE` - if this is defined then its value will be added as a header to all of the requests
+* `ZAP_AUTH_HEADER` - if this is defined then its value will be used as the header name - if it is not defined then the standard [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header will be used
+* `ZAP_AUTH_HEADER_SITE` - if this is defined then header will only be included in sites who’s name includes its value
+
+The env vars are standard operating system env vars so how you will need to set them will depend on your OS and possibly your shell.
+They need to be set before you start ZAP, they cannot be set once ZAP is already running.
+
+### Scripts
+
+[Scripts](/docs/desktop/start/features/scripts/) give you complete control over all of the requests and responses that are proxied through or generated by ZAP.
+
+If you are handling authentication yourself then HTTP Sender scripts are likely to be the most useful type.
+
+The [Community Scripts](https://github.com/zaproxy/community-scripts/) repo has a large number of 
+[HTTP Sender](https://github.com/zaproxy/community-scripts/tree/main/httpsender) scripts which you can use as the basis for your own solution.
+
+In order to use scripts in automation see the FAQ [How do you add a script to ZAP from the command line?](/faq/how-do-you-add-a-script-to-zap-from-the-command-line/).
+
+### Replacer Rules
+
+[Replacer Rules](/docs/desktop/addons/replacer/) are an easy way to replace and insert strings in requests and responses.
+
+The easiest way to create and test them is via the Desktop UI. You can still use them in automation - see the 
+FAQ [How do you find out what key to use to set a config value on the command line?](/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/) which includes an example for a Replacer rule.
```

Comment:
```suggestion
FAQ [How do you find out what key to use to set a config value on the command line?](/faq/how-do-you-find-out-what-key-to-use-to-set-a-config-value-on-the-command-line/) which includes an example for a Replacer rule being set via the CLI.
```

Reply:
Not sure about this - the link title explicitly mentions the command line so I dont think repeating that helps, and some people might think the CLI refers to zapcli :/